packages/components/src/internal/components/domainproperties/DomainForm.spec.tsx

```diff
@@ -684,4 +684,59 @@ describe('DomainForm', () => {
         expect(form).toMatchSnapshot();
         form.unmount();
     });
+
+    test('using allowImportExport', () => {
+        const domain = DomainDesign.create({});
+
+        const form = mount(
+            <DomainForm
+                domain={domain}
+                onChange={jest.fn()}
+                allowImportExport={true}
+            />
+        );
+
+        expect(form.find('.domain-form-manual-section').length).toEqual(1);
+        expect(form.find('.domain-form-manual-section').length).toEqual(1);
+        expect(form.find('.file-form-formats').text()).toContain('.json');
+
+        expect(form).toMatchSnapshot();
+        form.unmount();
+    });
+
+    test('using allowImportExport, field view', () => {
+        const fields = [];
+        fields.push({
+            name: 'key',
+            rangeURI: INT_RANGE_URI,
+            propertyId: 1,
+            propertyURI: 'test',
+        });
+
+        const domain = DomainDesign.create({
+            name: 'allowImportExport field view',
+            description: 'basic list domain form',
+            domainURI: 'test',
+            domainId: 1,
+            fields,
+            indices: [],
+        });
+
+        const form = mount(
+            <DomainForm
+                domain={domain}
+                onChange={jest.fn()}
+                domainFormDisplayOptions={{
+                    hideRequired: true,
+                }}
+                allowImportExport={true}
+            />
+        );
+
+        expect(form.find('.domain-toolbar-export-btn').length).toEqual(1);
```

Comment:
From your comment in the PR, it should be pretty easy to add a check here to see if the export button is enabled/disabled. See recent examples I added for button disabled checks here (around line 64): https://github.com/LabKey/labkey-ui-components/blob/release20.11-SNAPSHOT/packages/components/src/internal/components/domainproperties/OntologyLookupOptions.spec.tsx

Reply:
Added 'disabled' check in 'using allowImportExport, field view'. Would be nice to add a negative check for when there are no fields, but populating a DomainDesign with an empty field array gets us to the 'import file' view, and we'd have to do some mocking in order to click on a field row to delete it in order to get to the '0 fields' state.